tensorboard/webapp/core/views/page_title_container.ts

```diff
@@ -36,12 +58,44 @@ import {State} from '../state';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class PageTitleContainer {
-  readonly title$ = this.store.pipe(
-    select(getEnvironment),
-    map((env) => env.window_title || 'TensorBoard'),
-    // (it's an empty string when the `--window_title` flag is not set)
+  private readonly getExperimentId$ = this.store.select(getRouteKind).pipe(
+    filter((routeKind) => routeKind === RouteKind.EXPERIMENT),
+    withLatestFrom(this.store.select(getExperimentIdsFromRoute)),
+    map(([, experimentIds]) =>
+      experimentIds && experimentIds.length === 1 ? experimentIds[0] : null
+    )
+  );
+
+  private readonly experiment$ = this.getExperimentId$.pipe(
+    filter(Boolean),
+    mergeMap((experimentId) => {
+      return this.store.select(getExperiment, {experimentId});
+    })
+  );
+
+  private readonly experimentName$ = this.experiment$.pipe(
+    map((experiment) => (experiment ? experiment.name : null))
+  );
+
+  readonly title$ = this.store.select(getEnvironment).pipe(
+    combineLatestWith(this.experimentName$),
+    map(([env, experimentName]) => {
+      if (env.window_title && env.window_title.length > 0) {
+        // (it's an empty string when the `--window_title` flag is not set)
+        return env.window_title;
+      } else if (experimentName && experimentName.length > 0) {
+        return experimentName;
+      } else {
+        return this.tbServiceName
+          ? `TensorBoard.${this.tbServiceName}`
```

Comment:
Instead of making the service name only the part after "TensorBoard.", could we make it the entire string?  As a hypothetical example, something like "Vertex TensorBoard" might be desired one day, which does not begin with the "TensorBoard." prefix.

Reply:
+1 - also, for consistency maybe nice to refer to this as the "brand" (like `tbBrandName`) since that's the term we've used for the header logo text, historically:

https://github.com/tensorflow/tensorboard/blob/adb892c2dec51346ecaae69982b7c4162f5eddae/tensorboard/webapp/header/header_component.ts#L21

Reply:
Good point!